### PR TITLE
Update Operations Definition

### PR DESCRIPTION
**Pull request by spuniani**

### CODE REVIEW

File: src/main/java/com/sarality/db/UpdateOperation.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.sarality.db;
+
+/**
+ * Interface for defining updates to table schema from version to version
+ *
+ * @author Satya@ (Satya Puniani) on 01/03/17
+ */
+
+interface UpdateOperation {
```
Comment (abhideep) [inline src/main/java/com/sarality/db/UpdateOperation.java]: Can we call this something like SchemaUpdate and the Type as SchemaUpdateType.

As far as the interface goes, do we really need anything more that UpdateType and Db Version?

File: src/main/java/com/sarality/db/UpdateOperation.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  UpdateOperationType getUpdateType();
+
+  Column getExistingColumn();
+
+  Column getNewColumn();
```
Comment (abhideep) [inline src/main/java/com/sarality/db/UpdateOperation.java]: What does this mean for an AddTableSchemaUpdate?

File: src/main/java/com/sarality/db/UpdateOperationType.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.sarality.db;
+
+/**
+ * enum for list of operations to be supported when a database is upgraded.
+ *
+ * @author Satya@ (Satya Puniani) on 01/03/17
```
Comment (abhideep) [inline src/main/java/com/sarality/db/UpdateOperationType.java]: Can you please edit the new class template and remove the date for new classes that are generated

File: src/main/java/com/sarality/db/TableDefinition.java [hunk 4]
```diff
@@ -15,6 +20,8 @@
   private final String tableName;
   // The version of the database defined by the given set of columns
   private final int tableVersion;
+  // A list of updates for each version
+  private Map<Integer, Map<UpdateOperationType, List<UpdateOperation>>> versionUpdates = new HashMap<>();
```
Comment (abhideep) [inline src/main/java/com/sarality/db/TableDefinition.java]: Do we really need to do this processing in the TableDefinition. All this should have in a List of SchemaUpdate

The Map may need to be in a utility class like a SchemaUpdater, which in turn will reply on different SchemaUpdateSqlGenerators

**Comment by spuniani** [comment]
Made changes based on the review comments. @abhideep please review.

**Pull request merged**